framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,241 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Poly2TriMeshGenerator.h"
+
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/int_range.h"
+#include "libmesh/mesh_triangle_holes.h"
+#include "libmesh/parsed_function.h"
+#include "libmesh/poly2tri_triangulator.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", Poly2TriMeshGenerator);
+
+InputParameters
+Poly2TriMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  MooseEnum algorithm("BINARY EXHAUSTIVE", "BINARY");
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "boundary", "The MeshGenerator that defines the mesh outer boundary.");
+  params.addParam<unsigned int>(
+      "interpolate_boundary", 0, "How many more nodes to add in each outer boundary segment.");
+  params.addParam<bool>(
+      "refine_boundary", true, "Whether to allow automatically refining the outer boundary.");
+  params.addParam<bool>("smooth_triangulation",
+                        false,
+                        "Whether to do Laplacian mesh smoothing on the generated triangles.");
+  params.addParam<std::vector<MeshGeneratorName>>(
+      "holes", std::vector<MeshGeneratorName>(), "The MeshGenerators that define mesh holes.");
```

Comment:
Ugh I hate that this doesn't work

```suggestion
      "holes", {}, "The MeshGenerators that define mesh holes.");
```

Reply:
I tried removing the parameter entirely too (on the theory that that addParam API overload leaves values uninitialized but an "uninitialized" vector is the same as a default-initialized one), but that gives a runtime error.

Perhaps we could change the second `addParam` parameter from a `const S &` to a `T` (passing by value, then we move instead of copy later to keep the same efficiency)?  But that'd be quite an intrusive change I'd want to test in its own PR.

Reply:
Yeah - no worries. Just wanted to comment on my frustrations.